### ISSUE TITLE
Cannot use satcheljs along with regular mobx observable/actions

### ISSUE THREAD

**Issue by jair-cazarin** [Microsoft/satcheljs#36]
I've a few components using Mobx observables as local state rather than using setState as per this [post](https://medium.com/@mweststrate/3-reasons-why-i-stopped-using-react-setstate-ab73fc67a42e#.bttqjin7x).

I started moving app state to a satchel store but now mobx "regular" actions that are not going through satcheljs dispatcher are throwing because of the following code since spy is listening for any mobx action not just satcheljs actions.

```
spy((change) => {
    if (!getGlobalContext().inDispatch && change.type == "action") {
        throw new Error('The state may only be changed by a SatchelJS action.');
    }
});
```

it seems change.type == "action" is not enough to differentiate between mobx vs satchel actions

**Comment by jair-cazarin** [comment]
Also because satchel actions does not provide a way to decorate class methods I cannot force all actions to go through satchel dispatcher.

**Comment by jair-cazarin** [comment]
I guess the fix at the end of the day is to move the state to a satchel store and create proper satchel actions but still it feels like the two should work fine together.

**Comment by smikula** [comment]
You're right, this is something we need to fix.  When Satchel was conceived we wanted to be an opinionated framework for building an app on top of MobX.  (MobX is intentionally unopinionated, so we add a bunch of structure on top of it.)  However, this doesn't allow for the case where some component within an app is built on top of Satchel and other parts are using MobX directly.  We should introduce our own `useStrict()` API to disable this strict mode within Satchel.

Providing a way to decorate class methods is also a good point -- I've opened a separate issue (#38) for that.

**Comment by smikula** [comment]
This is fixed in v2.10.0.
